### DISCUSSION TITLE
web ui map image formats, compression levels, inherent quality and Download Size VS CPU load

### DISCUSSION THREAD

**Discussion by u-an-i** [albar965/littlenavmap#666] [Ideas]
web server supports an image format parameter. From "legacy ui" its assigned JPG also in "new ui".

it is accompanied by a quality parameter. Is this the compression indicator of the image format? JPG 30 -100 had no visible difference as far as i remember. Is it the compression value?

Additionally, Marble offers https://api.kde.org/marble/html/namespaceMarble.html#afb1c838f10a29)183408eeba218db4bfc (MapQuality) which has 4 relevant values of which "HighQuality" reads intruiging ("antialiasing for lines"). Can this be exposed to web ui? (eg. going low during zooming and going high or even print for individual refreshs (possibly even configurable for user for adjusting to their network and hardware performance)).

Evaluation Trade-Off:

bandwidth size && download time  VS  image generation processing time && server load better used for the sim :)

Where is / what would you consider the sweet spot ?

**Comment by albar965** [comment]
Image quality is JPEG compression. Marble quality levels are only to speed up movement of the map. Lower quality (no anti-alias, for example) is what you see when clicking into the map holding the mouse button. Only high and low are used. I see no use for outline and print does not make a difference as far as I can see.

Exposing this might make sense.

``` C++
  setMapQualityForViewContext(HighQuality, Still);
  setMapQualityForViewContext(LowQuality, Animation);
```

Edit: LNM also paints differently for Still and Animation depending on settings in options.
